python/graphframes/graphframe.py

```diff
@@ -102,6 +111,102 @@ def bfs(self, fromExpr, toExpr, edgeFilter=None, maxPathLength=10):
         jdf = builder.run()
         return DataFrame(jdf, self._sqlContext)
 
+    # Standard algorithms
+
+    def connected_components(self):
+        """
+        Computes the connected components of the graph.
+        :return:
+        """
+        jgf = self._jvm_graph.connectedComponents().run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def label_propagation(self, max_steps):
+        """
+        Runs static label propagation for detecting communities in networks.
+        :param max_steps: the number of super steps to be performed.
+        :return:
+        """
+        jgf = self._jvm_graph.labelPropagation().setMaxSteps(max_steps).run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def page_rank(self,
+                  reset_prob, source_id = None,
+                  fixed_num_iter = None,
+                  tolerance = None):
+        """
+        Runs the PageRank algorithm on the graph.
+        :param reset_prob:
+        :param source_id: (optional) the source vertex for a personalized PageRank.
+        :param fixed_num_iter: if set, the algorithm is run for a fixed number of iterations. In this case, the
+               `tolerance` parameter is ignored.
+        :param tolerance: if set, the algorithm is run until the given tolerance. In this case, the `fixed_num_iter`
```

Comment:
This documentation does not match the implementation.  (This raises an exception if both are set.)